python_modules/dagster/dagster/core/system_config/objects.py

```diff
@@ -152,18 +167,28 @@ def build(
         run_config_schema = pipeline_def.get_run_config_schema(mode)
 
         if run_config_schema.config_mapping:
-            outer_evr = process_config(
-                run_config_schema.config_mapping.config_schema.config_type,
-                run_config,
-            )
+            if run_config_schema.config_mapping.enable_pre_processing:
+                outer_evr = process_config(
+                    run_config_schema.config_mapping.config_schema.config_type,
+                    run_config,
+                )
+            else:
+                outer_evr = validate_config(
+                    run_config_schema.config_mapping.config_schema.config_type,
+                    run_config,
+                )
             if not outer_evr.success:
                 raise DagsterInvalidConfigError(
                     f"Error in config mapping for pipeline {pipeline_def.name} mode {mode}",
                     outer_evr.errors,
                     run_config,
                 )
+            outer_config = resolve_defaults(
+                cast("ConfigType", run_config_schema.config_mapping.config_schema.config_type),
+                outer_evr.value,
+            ).value
```

Comment:
we are double resolving defaults int he process_config case right? seems like this belongs in the if block 

might make sense to make an entry point like `process_config` that captures the validate then resolve defaults pattern